### PR TITLE
CAPV image stamper

### LINKED ISSUE #3: Import CAPV implementation
@akutz Could you please import the CAPV implementation?

/assign @akutz

### PR DESCRIPTION
**Pull request by akutz**
This PR fixes #3 and imports the CAPV image stamper by sub-tree adding the CAPV `build/images` directory into the `images/capv` directory of this project.

cc @detiber Here's how I had to do this since the projects were unrelated and the source was a sub-directory of its repository:

1. Start in the CAPV repository:

    ```shell
    cd $GOPATH/src/sigs.k8s.io/cluster-api-provider-vsphere
    ```

2. Split the source directory in CAPV into its own branch:

    ```shell
    git subtree split -b subtree/build-images \
                      -P build/images \
                      feature/mv-machine-image-scripts
    ```

    The above command:

    * `-b subtree/build-images` - Creates a new branch, `subtree/build-images`
    * `-P build/images` - The contents of the `build/images` directory become the root of the repository in the new branch
    * `feature/mv-machine-image-scripts` - The ref from which to do the split

3. Switch to the image-builder repository:

    ```shell
    cd $GOPATH/src/sigs.k8s.io/image-builder
    ```

4. Add the local CAPV repo as a remote for the image-builder repo:

    ```shell
    git remote add capv $GOPATH/src/sigs.k8s.io/cluster-api-provider-vsphere
    ```

5. Remove the placeholder `images/capv` directory and create a commit:

    ```shell
    rm -fr images/capv && git add -u && git commit
    ```

6. Add the CAPV image tooling to `images/capv` via a subtree add:

    ```shell
    git subtree add \
      --squash \
      -m "Subtree merge CAPV image/builds" \
      -P images/capv/ \
      capv \
      subtree/build-images
    ```

    The above command:

    * `--squash` - Squash the commits from the source repository. Since the two repositories do not share a common merge-base, performing a non-squashed add causes the history of `image-builder` to be rebased. Its initial commit is no longer the initial commit. I opted to avoid this.
    * `-m "Subtree merge CAPV image/builds"` - Provide a nice message for the merge commit, otherwise it's not apparent what was done.
    * `-P images/capv` - Indicate to what sub-directory to read the source data into
    * `capv` - The name of the remote from which to read
    * `subtree/build-images` - The name of the ref on the remote to read

### CODE REVIEW

**Comment by timothysc** [comment]
Before I dig, is this really capv specific or did you generalize it to be useful across capa?

**Comment by akutz** [comment]
Hi @timothysc,

It's completely useful across cap*. All the enhancements are CAPV-agnostic. There's CAPV-specific bits, but that's just compatibility issues. I just pushed the [subtree split](https://github.com/akutz/cluster-api-provider-vsphere/tree/subtree/build-images) so you can look just at the changes specific to the image tooling.

**Comment by akutz** [comment]
Hi @timothysc,

I'm happy to walk you through the major bits:

* CAPA doesn't actually build from ISO, CAPV must do so. Therefore there are Debian seed and RHEL kickstart files (as well as associated boot commands) to build images from scratch
* Can install Kubernetes from a Google-compatible bucket (kubernetes-release/release-dev)
* Fixes missing steps in doing sysprep ahead of sealing the image
* Enables local testing

File: images/capv/cloudinit/id_rsa.capi [hunk 1]
```diff
@@ -0,0 +1,27 @@
+-----BEGIN OPENSSH PRIVATE KEY-----
```
Comment (timothysc) [inline images/capv/cloudinit/id_rsa.capi]: This seems very odd to be posting a key to a public repo vs. either generating the key, or having a user specified variable.

Comment (akutz) [reply images/capv/cloudinit/id_rsa.capi]: Hi @timothysc,

I encourage you to take a look first at the README, specifically the [`cloudinit` section](https://github.com/akutz/cluster-api-provider-vsphere/tree/subtree/build-images#the-cloudinit-directory):

> The cloudinit directory contains files that:
>
> * Are example data used for testing
> * Are not included in any of the images
> * Should not be used in production systems

**Comment by akutz** [comment]
> @akutz
> Could you shuffle the directory to remove the (capv) to capi?

Hi @timothysc,

I am not sure what you mean. I placed the files in the placeholder directory. Are you requesting that I remove the `images/capa` placeholder and rename `images/capv` to `images/capi`?

**Comment by timothysc** [comment]
> are you requesting that I remove the images/capa placeholder and rename images/capv to images/capi 

Yes please.

**Comment by akutz** [comment]
> > are you requesting that I remove the images/capa placeholder and rename images/capv to images/capi
> 
> Yes please.

Done and done.

**Comment by akutz** [comment]
> We should still remove the private key imo and allow user option but I'll open an issue for that.

Hi @timothysc,

I disagree. The entire point of that directory is to have a fully baked cloud-init payload that may be used to immediately test images. It's documented as such.

**Pull request merged**